cmd/promql-compliance-tester/main.go

```diff
@@ -96,8 +96,13 @@ func main() {
 	}
 
 	// Expand all placeholder variations in the templated test cases.
-	end := time.Now().Add(-2 * time.Minute)
-	start := end.Add(-10 * time.Minute)
+	var start time.Time
+	if cfg.Start > 0 {
+		start = time.Unix(cfg.Start, 0)
```

Comment:
Cool! Would you be up for doing parsing like here: https://github.com/prometheus/prometheus/blob/main/cmd/promtool/main.go#L693-L702

Being able to pass in also sub-second timestamps is important because otherwise everything ends up aligned at full second boundaries (since the resolution step is currently also just a multiple of seconds), potentially not catching some implementations which incorrectly always align query timestamps to seconds.

Reply:
Actually, I think from my Prometheus habits perspective, it would be more intuitive to set the end time, not the start time, since that's how the Prometheus UI works as well (you pick an end time for the graph, and a range for how far back it should go).

Reply:
(In the end, we should make all of `start` / `end` / `resolution` all configurable)

Reply:
That all sounds good to me. I implemented everything you suggested here.